src/circuit/gadgets/constraints.rs

```diff
@@ -711,4 +761,52 @@ mod tests {
             assert!(cs.is_satisfied());
         }
     }
+
+    #[test]
+    // needs to return Result because the macros use ?.
+    fn test_and_or_v() -> Result<(), SynthesisError> {
+        use rand::Rng;
```

Comment:
Can we use proptest here?

Reply:
Yeah, I tried and couldn't quite figure it out. I'll see if I can copy from that now.